pootle/apps/pootle_store/models.py

```diff
@@ -1864,6 +1877,20 @@ def insert(i1, j1, j2, offset):
                     self.get_max_unit_revision())
                 )
 
+    def increment_unsynced_unit_revision(self, update_revision):
+        filter_by = {
+            'store': self,
+            'revision__gt': self.last_sync_revision,
+            'revision__lt': update_revision,
+        }
+        count = 0
+        for unit in Unit.simple_objects.filter(**filter_by) \
+                        .exclude(state=OBSOLETE):
```

Comment:
shorter but not better 

In [7]: list(Unit.simple_objects.filter(store=s).all())
[28/Jul/2015 09:54:18]	(6.772) SELECT `pootle_store_unit`.`id`, `pootle_store_unit`.`store_id`, `pootle_store_unit`.`index`, `pootle_store_unit`.`unitid`, `pootle_store_unit`.`unitid_hash`, `pootle_store_unit`.`source_f`, `pootle_store_unit`.`source_hash`, `pootle_store_unit`.`source_wordcount`, `pootle_store_unit`.`source_length`, `pootle_store_unit`.`target_f`, `pootle_store_unit`.`target_wordcount`, `pootle_store_unit`.`target_length`, `pootle_store_unit`.`developer_comment`, `pootle_store_unit`.`translator_comment`, `pootle_store_unit`.`locations`, `pootle_store_unit`.`context`, `pootle_store_unit`.`state`, `pootle_store_unit`.`revision`, `pootle_store_unit`.`creation_time`, `pootle_store_unit`.`mtime`, `pootle_store_unit`.`submitted_by_id`, `pootle_store_unit`.`submitted_on`, `pootle_store_unit`.`commented_by_id`, `pootle_store_unit`.`commented_on`, `pootle_store_unit`.`reviewed_by_id`, `pootle_store_unit`.`reviewed_on` FROM `pootle_store_unit` INNER JOIN `pootle_store_store` ON ( `pootle_store_unit`.`store_id` = `pootle_store_store`.`id` ) WHERE `pootle_store_unit`.`store_id` = 24394 ORDER BY `pootle_store_store`.`pootle_path` ASC, `pootle_store_unit`.`index` ASC; args=(24394,)

In [4]: list(s.unit_set.all())
[28/Jul/2015 09:52:07]	(13.082) SELECT `pootle_store_unit`.`id`, `pootle_store_unit`.`store_id`, `pootle_store_unit`.`index`, `pootle_store_unit`.`unitid`, `pootle_store_unit`.`unitid_hash`, `pootle_store_unit`.`source_f`, `pootle_store_unit`.`source_hash`, `pootle_store_unit`.`source_wordcount`, `pootle_store_unit`.`source_length`, `pootle_store_unit`.`target_f`, `pootle_store_unit`.`target_wordcount`, `pootle_store_unit`.`target_length`, `pootle_store_unit`.`developer_comment`, `pootle_store_unit`.`translator_comment`, `pootle_store_unit`.`locations`, `pootle_store_unit`.`context`, `pootle_store_unit`.`state`, `pootle_store_unit`.`revision`, `pootle_store_unit`.`creation_time`, `pootle_store_unit`.`mtime`, `pootle_store_unit`.`submitted_by_id`, `pootle_store_unit`.`submitted_on`, `pootle_store_unit`.`commented_by_id`, `pootle_store_unit`.`commented_on`, `pootle_store_unit`.`reviewed_by_id`, `pootle_store_unit`.`reviewed_on`, `pootle_store_store`.`id`, `pootle_store_store`.`file`, `pootle_store_store`.`parent_id`, `pootle_store_store`.`translation_project_id`, `pootle_store_store`.`pootle_path`, `pootle_store_store`.`name`, `pootle_store_store`.`file_mtime`, `pootle_store_store`.`state`, `pootle_store_store`.`creation_time`, `pootle_store_store`.`last_sync_revision`, `pootle_store_store`.`obsolete`, `accounts_user`.`id`, `accounts_user`.`password`, `accounts_user`.`last_login`, `accounts_user`.`username`, `accounts_user`.`email`, `accounts_user`.`full_name`, `accounts_user`.`is_active`, `accounts_user`.`is_superuser`, `accounts_user`.`date_joined`, `accounts_user`.`unit_rows`, `accounts_user`.`rate`, `accounts_user`.`review_rate`, `accounts_user`.`hourly_rate`, `accounts_user`.`score`, `accounts_user`.`currency`, `accounts_user`.`is_employee`, `accounts_user`.`twitter`, `accounts_user`.`website`, `accounts_user`.`linkedin`, `accounts_user`.`bio`, T4.`id`, T4.`password`, T4.`last_login`, T4.`username`, T4.`email`, T4.`full_name`, T4.`is_active`, T4.`is_superuser`, T4.`date_joined`, T4.`unit_rows`, T4.`rate`, T4.`review_rate`, T4.`hourly_rate`, T4.`score`, T4.`currency`, T4.`is_employee`, T4.`twitter`, T4.`website`, T4.`linkedin`, T4.`bio`, T5.`id`, T5.`password`, T5.`last_login`, T5.`username`, T5.`email`, T5.`full_name`, T5.`is_active`, T5.`is_superuser`, T5.`date_joined`, T5.`unit_rows`, T5.`rate`, T5.`review_rate`, T5.`hourly_rate`, T5.`score`, T5.`currency`, T5.`is_employee`, T5.`twitter`, T5.`website`, T5.`linkedin`, T5.`bio` FROM `pootle_store_unit` INNER JOIN `pootle_store_store` ON ( `pootle_store_unit`.`store_id` = `pootle_store_store`.`id` ) LEFT OUTER JOIN `accounts_user` ON ( `pootle_store_unit`.`submitted_by_id` = `accounts_user`.`id` ) LEFT OUTER JOIN `accounts_user` T4 ON ( `pootle_store_unit`.`commented_by_id` = T4.`id` ) LEFT OUTER JOIN `accounts_user` T5 ON ( `pootle_store_unit`.`reviewed_by_id` = T5.`id` ) WHERE `pootle_store_unit`.`store_id` = 24394 ORDER BY `pootle_store_store`.`pootle_path` ASC, `pootle_store_unit`.`index` ASC; args=(24394,)